staging/src/k8s.io/apiserver/pkg/cel/image.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+)
+
+var (
+	ImageType = cel.ObjectType("kubernetes.Image")
+)
+
+type ImageReference struct {
+	Image      string `json:"image,omitempty"`
```

Comment:
we should probably document these?

Reply:
Also, I'm curious why this is a JSON serializable type.  What's the use case for this?  I was expecting `.String()` on `ImageReference` to the the way this is typically serialized..